recipes/libmysqlclient/all/conanfile.py

```diff
@@ -100,160 +75,168 @@ def loose_lt_semver(v1, v2):
+        replace_in_file(self, os.path.join(self.source_folder, "cmake", "install_macros.cmake"),
+                        "  INSTALL_DEBUG_SYMBOLS(",
+                        "  # INSTALL_DEBUG_SYMBOLS(")
+
+    def generate(self):
+        cmake = CMakeToolchain(self)
+        # Not used anywhere in the CMakeLists
+
+        cmake.cache_variables["DISABLE_SHARED"] = not self.options.shared
+        cmake.cache_variables["STACK_DIRECTION"] = "-1"  # stack grows downwards, on very few platforms stack grows upwards
+        cmake.cache_variables["WITHOUT_SERVER"] = True
+        cmake.cache_variables["WITH_UNIT_TESTS"] = False
+        cmake.cache_variables["ENABLED_PROFILING"] = False
+        cmake.cache_variables["MYSQL_MAINTAINER_MODE"] = False
+        cmake.cache_variables["WIX_DIR"] = False
+
+        cmake.cache_variables["WITH_LZ4"] = "system"
+
+        cmake.cache_variables["WITH_ZSTD"] = "system"
+        cmake.cache_variables["ZSTD_INCLUDE_DIR"] = self.dependencies["zstd"].cpp_info.includedirs[0]
 
         if is_msvc(self):
-            cmake.definitions["WINDOWS_RUNTIME_MD"] = "MD" in msvc_runtime_flag(self)
+            cmake.cache_variables["WINDOWS_RUNTIME_MD"] = "MD" in msvc_runtime_flag(self)
 
-        cmake.definitions["WITH_SSL"] = self.deps_cpp_info["openssl"].rootpath
+        cmake.cache_variables["WITH_SSL"] = self.dependencies["openssl"].package_folder
 
-        cmake.definitions["WITH_ZLIB"] = "system"
-        cmake.configure(source_dir=self._source_subfolder)
-        return cmake
+        cmake.cache_variables["WITH_ZLIB"] = "system"
+        cmake.generate()
 
     def build(self):
         self._patch_files()
-        cmake = self._configure_cmake()
-        with tools.run_environment(self):
-            cmake.build()
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
 
     def package(self):
-        cmake = self._configure_cmake()
-        with tools.run_environment(self):
-            cmake.install()
-        os.mkdir(os.path.join(self.package_folder, "licenses"))
+        cmake = CMake(self)
+        cmake.install()
+        mkdir(self, os.path.join(self.package_folder, "licenses"))
         rename(self, os.path.join(self.package_folder, "LICENSE"), os.path.join(self.package_folder, "licenses", "LICENSE"))
-        os.remove(os.path.join(self.package_folder, "README"))
+        rm(self, "README", self.package_folder)
         rm(self, "*.pdb", self.package_folder, recursive=True)
         rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
         rmdir(self, os.path.join(self.package_folder, "docs"))
         rmdir(self, os.path.join(self.package_folder, "share"))
         if self.settings.os == "Windows" and self.options.shared:
-            self.copy("*.dll", "bin", keep_path=False)
+            copy("*.dll", src=self.package_folder, dst="bin", keep_path=False)
```

Comment:
mmm I think here the first argument needs to be self, the `src` is likely `self.build_folder` and the `dst` is `os.path.join(self.package_folder, "bin").

in conan 1.x self.copy is contextual which caused confusion :D